### PR TITLE
modify update arg based on Nix version

### PR DESCRIPTION
**Pull request by NotAShelf**
Modifies the update command arguments depending on the current nix version.

### CODE REVIEW

**Comment by NotAShelf** [comment]
rawdogged this a little, will revise in a bit.

**Comment by viperML** [comment]
for src/utils.rs, you can use a `eyre::Result<T>` return type instead.
https://docs.rs/eyre/latest/eyre/#details

**Comment by NotAShelf** [comment]
I never used eyre before, hope this is what you mean. If it's not, then I'll admit I was completely clueless.

File: src/util.rs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+/// # Arguments
+///
+/// * `current` - A string slice representing the current version.
+/// * `target` - A string slice representing the target version to compare against.
+///
+/// # Returns
+///
+/// * `Result<std::cmp::Ordering>` - The comparison result.
+pub fn compare_semver(current: &str, target: &str) -> Result<std::cmp::Ordering> {
+    let current = Version::parse(current).unwrap();
```
Comment (viperML) [inline src/util.rs]: Also use ? instead of unwrap here

**Comment by viperML** [comment]
haven't tested but lgtm

**Comment by NotAShelf** [comment]
tested with nix 2.18.1 and nix super 2.20.0 before the latest changes. will test again shortly

**Comment by NotAShelf** [comment]
**Output for Nix Super (2.20.0)**

<details>
```sh
$ nix --version 
nix (Nix Super) 2.20.0pre20240216_924eb11
        
$ cargo run -- os boot --update --verbose
DEBUG nh:21: args=NHParser { verbose: true, command: Os(OsArgs { action: Boot(OsRebuildArgs { common: CommonRebuildArgs { dry: false, ask: false, flakeref: FlakeRef("/home/notashelf/.config/nyx"), update: true, no_nom: false }, hostname: None, specialisation: None, no_specialisation: false, extra_args: [] }) }) }
DEBUG nh::nixos:42: out_dir: TempDir { path: "/tmp/nh-os-CWQLoX" }
DEBUG nh::nixos:43: out_link "/tmp/nh-os-CWQLoX/result"
DEBUG nh::nixos:69: nix_version: "2.20.0"
DEBUG nh::nixos:70: update_args: ["nix", "flake", "update", "--flake", "/home/notashelf/.config/nyx"]
> Updating flake
DEBUG nh::commands:53: cmd=Exec { nix flake update --flake /home/notashelf/.config/nyx }
[39.6/0.6 MiB DL] downloading 'https://api.github.com/repos/hyprwm/Hyprland/tarball/3e67ee0f5f66c74c0a0bd4f797361fd100c2e2d2'^C  
```
</details>

**Output for Nix Super (2.20.0)**

<details>
```sh
$ nix --version 
nix (Nix Super) 2.20.0pre20240216_924eb11
        
$ cargo run -- os boot --update --verbose
DEBUG nh:21: args=NHParser { verbose: true, command: Os(OsArgs { action: Boot(OsRebuildArgs { common: CommonRebuildArgs { dry: false, ask: false, flakeref: FlakeRef("/home/notashelf/.config/nyx"), update: true, no_nom: false }, hostname: None, specialisation: None, no_specialisation: false, extra_args: [] }) }) }
DEBUG nh::nixos:42: out_dir: TempDir { path: "/tmp/nh-os-CWQLoX" }
DEBUG nh::nixos:43: out_link "/tmp/nh-os-CWQLoX/result"
DEBUG nh::nixos:69: nix_version: "2.20.0"
DEBUG nh::nixos:70: update_args: ["nix", "flake", "update", "--flake", "/home/notashelf/.config/nyx"]
> Updating flake
DEBUG nh::commands:53: cmd=Exec { nix flake update --flake /home/notashelf/.config/nyx }
[39.6/0.6 MiB DL] downloading 'https://api.github.com/repos/hyprwm/Hyprland/tarball/3e67ee0f5f66c74c0a0bd4f797361fd100c2e2d2'^C  
```
</details>

**Output for Nix (unstable branch) (2.18.1)**

<details>
$ nix --version
nix (Nix) 2.18.1

$ cargo run -- os boot --update --verbose
TRACE nh::logging:86: Logging OK
DEBUG nh:21: args=NHParser { verbose: true, command: Os(OsArgs { action: Boot(OsRebuildArgs { common: CommonRebuildArgs { dry: false, ask: false, flakeref: FlakeRef("/home/notashelf/.config/nyx"), update: true, no_nom: false }, hostname: None, specialisation: None, no_specialisation: false, extra_args: [] }) }) }
DEBUG nh::nixos:42: out_dir: TempDir { path: "/tmp/nh-os-SAWrdy" }
DEBUG nh::nixos:43: out_link "/tmp/nh-os-SAWrdy/result"
DEBUG nh::nixos:69: nix_version: "2.18.1"
DEBUG nh::nixos:70: update_args: ["nix", "flake", "update", "/home/notashelf/.config/nyx"]
> Updating flake
DEBUG nh::commands:53: cmd=Exec { nix flake update /home/notashelf/.config/nyx }
warning: unknown experimental feature 'configurable-impure-env'
warning: unknown experimental feature 'git-hashing'
warning: unknown experimental feature 'verified-fetches'
[5.3 MiB DL] downloading 'https://api.github.com/repos/hyprwm/Hyprland/tarball/3e67ee0f5f66c74c0a0bd4f797361fd100c2e2d2'^C
</details>

Seems to be working as intended.

**Comment by viperML** [comment]
should I merge?

**Comment by NotAShelf** [comment]
should be goo to merge

**Pull request merged**